### ISSUE TITLE
add preview images for examples which don't have them 

### ISSUE THREAD

**Issue by ofTheo** [openframeworks/openFrameworks#7796]
This will be helpful for the gallery page: https://openframeworks.cc/examples/ 

Images should be in the following format: 

```
examples/math/myExampleName/myExampleName.png
examples/math/myExampleName/myExampleName.gif 
```

related: #7790

**Comment by artificiel** [comment]
are jpg allowed?

**Comment by artificiel** [comment]
i am also wondering if this is not the occasion to normalize the example name scheme? ex: most end with "Example", but not all. 
```
list of not found example images (excluding ios, tvos and android)

MISSING EXAMPLE IMAGE: examples/3d/advanced3dExample
MISSING EXAMPLE IMAGE: examples/3d/assimp3DModelLoaderExample but found candidate assimpExample.gif
MISSING EXAMPLE IMAGE: examples/3d/cameraLensOffsetExample
MISSING EXAMPLE IMAGE: examples/3d/cameraParentingExample
MISSING EXAMPLE IMAGE: examples/3d/meshFromCameraExample
MISSING EXAMPLE IMAGE: examples/3d/normalsExample
MISSING EXAMPLE IMAGE: examples/3d/quaternionArcballExample but found candidate quaternionArcBallExample.png
MISSING EXAMPLE IMAGE: examples/3d/quaternionLatLongExample but found candidate quaternionLongLatExample.png
MISSING EXAMPLE IMAGE: examples/communication/networkTcpClientExample
MISSING EXAMPLE IMAGE: examples/communication/networkTcpServerExample
MISSING EXAMPLE IMAGE: examples/communication/networkUdpReceiverExample
MISSING EXAMPLE IMAGE: examples/communication/networkUdpSenderExample
MISSING EXAMPLE IMAGE: examples/communication/oscChatSystemExample
MISSING EXAMPLE IMAGE: examples/communication/oscOptionalExample
MISSING EXAMPLE IMAGE: examples/communication/oscReceiveExample
MISSING EXAMPLE IMAGE: examples/communication/oscSenderExample
MISSING EXAMPLE IMAGE: examples/computer_vision/kinectExample
MISSING EXAMPLE IMAGE: examples/computer_vision/opencvHaarFinderExample
MISSING EXAMPLE IMAGE: examples/computer_vision/opencvImageClassification
MISSING EXAMPLE IMAGE: examples/computer_vision/opencvPeopleDetection
MISSING EXAMPLE IMAGE: examples/events/notifyEventExample
MISSING EXAMPLE IMAGE: examples/events/rpiTouchExample but found candidate rpiTouchExampleScreenshot.jpg
MISSING EXAMPLE IMAGE: examples/events/simpleTimerEventExample but found candidate simpleTimer.png
MISSING EXAMPLE IMAGE: examples/gl/materialPBR but found candidate materialPBRExample.png
MISSING EXAMPLE IMAGE: examples/gl/materialPBRAdvanced but found candidate materialPBRAdvancedExample.png
MISSING EXAMPLE IMAGE: examples/gl/pixelBufferExample
MISSING EXAMPLE IMAGE: examples/gl/pointsAsTexturesExample
MISSING EXAMPLE IMAGE: examples/gl/shaderExample
MISSING EXAMPLE IMAGE: examples/gl/slowFastRenderingExample
MISSING EXAMPLE IMAGE: examples/gl/textureBufferInstancedExample
MISSING EXAMPLE IMAGE: examples/gl/textureExample
MISSING EXAMPLE IMAGE: examples/gl/textureScreengrabExample
MISSING EXAMPLE IMAGE: examples/gl/threadedPixelBufferExample
MISSING EXAMPLE IMAGE: examples/gl/transformFeedbackExample
MISSING EXAMPLE IMAGE: examples/gl/vboExample
MISSING EXAMPLE IMAGE: examples/gl/vboMeshDrawInstancedExample
MISSING EXAMPLE IMAGE: examples/gl/viewportExample
MISSING EXAMPLE IMAGE: examples/gles/customEGLWindowSettingsExample
MISSING EXAMPLE IMAGE: examples/graphics/colorExample but found candidate color.png
MISSING EXAMPLE IMAGE: examples/graphics/colorsExtendedExample but found candidate colorsExtended.png
MISSING EXAMPLE IMAGE: examples/graphics/fontsExample
MISSING EXAMPLE IMAGE: examples/graphics/imageSubsectionExample
MISSING EXAMPLE IMAGE: examples/graphics/polylineBlobsExample
MISSING EXAMPLE IMAGE: examples/graphics/rectangleAlignmentAndScalingExample but found candidate rectangleAlignmentAndScaling.png
MISSING EXAMPLE IMAGE: examples/graphics/simpleColorKeyExample but found candidate simpleColorKey.png
MISSING EXAMPLE IMAGE: examples/graphics/vectorGraphicsExample
MISSING EXAMPLE IMAGE: examples/gui/guiExample but found candidate gui.gif
MISSING EXAMPLE IMAGE: examples/gui/guiFromParametersExample
MISSING EXAMPLE IMAGE: examples/gui/oscParametersReceiverExample
MISSING EXAMPLE IMAGE: examples/gui/oscParametersSenderExample
MISSING EXAMPLE IMAGE: examples/gui/parameterEdgeCasesExample
MISSING EXAMPLE IMAGE: examples/gui/parameterGroupExample
MISSING EXAMPLE IMAGE: examples/input_output/fileOpenSaveDialogExample but found candidate fileOpenSaveDialog.png
MISSING EXAMPLE IMAGE: examples/input_output/imageSaverExample but found candidate imageSaverExample2.png
MISSING EXAMPLE IMAGE: examples/input_output/pugiXmlExample
MISSING EXAMPLE IMAGE: examples/input_output/svgExample
MISSING EXAMPLE IMAGE: examples/input_output/systemSpeakExample but found candidate systemSpeak.png
MISSING EXAMPLE IMAGE: examples/input_output/xmlSettingsExample
MISSING EXAMPLE IMAGE: examples/math/noiseField2dExample
MISSING EXAMPLE IMAGE: examples/math/randomExample
MISSING EXAMPLE IMAGE: examples/shader/01_simpleColorQuad but found candidate 01_simpleColorQuadExample.png
MISSING EXAMPLE IMAGE: examples/shader/04_simpleTexturing
MISSING EXAMPLE IMAGE: examples/shader/05_alphaMasking
MISSING EXAMPLE IMAGE: examples/shader/06_multiTexture
MISSING EXAMPLE IMAGE: examples/shader/08_displacementMap
MISSING EXAMPLE IMAGE: examples/shader/09_gaussianBlurFilter but found candidate gaussianBlurFilterExample.png
MISSING EXAMPLE IMAGE: examples/sound/audioOutputExample
MISSING EXAMPLE IMAGE: examples/strings/conversionExample
MISSING EXAMPLE IMAGE: examples/strings/ofLogExample but found candidate loggingExample.png
MISSING EXAMPLE IMAGE: examples/threads/threadChannelExample
MISSING EXAMPLE IMAGE: examples/threads/threadedImageLoaderExample
MISSING EXAMPLE IMAGE: examples/tvOS/emptyExample
MISSING EXAMPLE IMAGE: examples/tvOS/ofBoxExample
MISSING EXAMPLE IMAGE: examples/video/slitscanRadialClockExample but found candidate radial-push-ribbon.png
MISSING EXAMPLE IMAGE: examples/windowing/noWindowExample
```

**Comment by ofTheo** [comment]
@artificiel yes def - be great to standardize all that don't end in example and generate the thumbs ( could even be automated maybe? ).  

jpg is allowed too and keeping the file name the same as the example folder name will make the website demo page etc much easier to automate.

**Comment by ofTheo** [comment]
@artificiel would you want to take a stab at renaming missing images and/or generating new ones?

**Comment by artificiel** [comment]
the task is not trivial to complete properly once started (a couple of hours I guess) and I'm in an async/sporadic availability period. I can't commit to that short term — and for sure we must make sure everything is done correctly for such as "pass". 

looks like:

 - rename the examples to spec;
 - rename images or generate fresh ones (which requires a compile/run/grab pass);
 - if renamed, verify/touch up the READMEs (as most link to the image);
  
about automation, would it be envisionable to have a compile-time flag that would enable an `ofSaveFrame` sequence? it could be an expansion of compile/runAllExamples.sh, which could harvest the screenshots and ffmpeg a gif. not all examples have movement without user interaction, but it would be a start. 

somewhere in `ofMainLoop::loopOnce()`:
```c++
#ifdef AUTOGRABBER_ENABLED
			static size_t grabbed {0};
			if (ofGetFrameNum()%60==30) { // skips early frames
				ofSaveFrame();
				if (++grabbed == 15) exit();
			}
#endif
```

**Comment by NickHardeman** [comment]
@ofTheo @artificiel I started on this, but will take me a little bit to complete. There are a lot of examples! I have just been doing it manually for now. :) 
I have also been checking which compile with Emscipten and I can start a list. And have already found a few bugs I made PRs for.

**Comment by ofTheo** [comment]
closed by #7801
